Library/Homebrew/cmd/list.rb

```diff
@@ -69,45 +76,46 @@ def list_args
   def list
     args = list_args.parse
 
-    return list_casks(args: args) if args.cask?
-
-    if args.unbrewed?
-      raise UsageError, "`--unbrewed` does not take a formula/cask argument." unless args.no_named?
-
-      return list_unbrewed
-    end
-
     # Unbrewed uses the PREFIX, which will exist
     # Things below use the CELLAR, which doesn't until the first formula is installed.
     unless HOMEBREW_CELLAR.exist?
-      raise NoSuchKegError, args.named.first if args.named.present?
+      raise NoSuchKegError, args.named.first if args.named.present? && !args.cask?
 
       return
     end
 
-    if args.pinned? || args.versions?
+    if args.full_name?
+      unless args.cask?
+        formula_names = args.no_named? ? Formula.installed : args.named.to_resolved_formulae
+        full_formula_names = formula_names.map(&:full_name).sort(&tap_and_name_comparison)
+        full_formula_names = Formatter.columns(full_formula_names) unless args.public_send(:'1?')
+        puts full_formula_names unless full_formula_names.blank?
+      end
+      unless args.formula? || (!args.no_named? && !args.cask?)
```

Comment:
@EricFromCanada Can this be refactored to not use `unless ... ||` which is hard to follow.